### ISSUE TITLE
symlink-info gives the wrong output for relative paths

### ISSUE THREAD

**Issue by wjandrea** [wjandrea/what-bash#1]
If `$link_deref` is relative, it assumes `$path` is relative, which is incorrect.

How I noticed it:

```
$ cd /usr/src/linux-headers-5.4.0-104-generic/scripts
$ symlink-info extract-vmlinux
extract-vmlinux
    symlink: ../../linux-headers-5.4.0-104/scripts/extract-vmlinux
    canonical path: /usr/src/linux-headers-5.4.0-104/linux-headers-5.4.0-104/scripts/extract-vmlinux
```

The canonical path is wrong. To get the correct one:

```
$ readlink -e -- extract-vmlinux 
/usr/src/linux-headers-5.4.0-104/scripts/extract-vmlinux
```

**Comment by wjandrea** [comment]
Problem seems to be in line 121:

https://github.com/wjandrea/what-bash/blob/46d859360bdd04ab196c548c7f1f21e30b33583f/symlink-info.sh#L115-L122

This doesn't account for `$path` as a barename, i.e. not containing a slash.

**Comment by wjandrea** [comment]
Closed in 5b8e0ef
